pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceProducer.java

```diff
@@ -597,7 +634,11 @@ private static void runProducer(int producerId,
             }
             // Send messages on all topics/producers
             long totalSent = 0;
+            AtomicReference<Transaction> transactionAtomicReference = buildTransaction(client, arguments);
+            AtomicLong numMessageSend = new AtomicLong(0);
+            Semaphore semaphore = new Semaphore(arguments.numMessagesPerTransaction);
             while (true) {
+                Transaction transaction = transactionAtomicReference.get();
```

Comment:
only enable transaction need this right?

Reply:
If transaction is disable, there transaction will be null without exception.  In terms of results, it is no different from adding a judgment. I didn’t add that judgment in order to simplify

Reply:
But now it seems that it has lost its legibility and is very unfriendly to code readers